src/hotspot/share/services/heapDumper.cpp

```diff
@@ -1512,6 +1512,45 @@ class ClassDumper : public KlassClosure {
   }
 };
 
+// Support class used to generate HPROF_LOAD_CLASS records
+
+class LoadedClassDumper : public LockedClassesDo {
+ private:
+  AbstractDumpWriter* _writer;
+  GrowableArray<Klass*>* _klass_map;
+  u4 _class_serial_num;
+  AbstractDumpWriter* writer() const { return _writer; }
+  void add_class_serial_number(Klass* k, int serial_num) {
+      _klass_map->at_put_grow(serial_num, k);
+  }
+ public:
+  LoadedClassDumper(AbstractDumpWriter* writer, GrowableArray<Klass*>* klass_map)
+    : _writer(writer), _klass_map(klass_map), _class_serial_num(0) {}
+  void do_klass(Klass* k);
+};
+
+void LoadedClassDumper::do_klass(Klass* k) {
```

Comment:
Any reason not to merge this definition straight in declaration?

Reply:
Merged.